arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/TerminalOps.kt

```diff
@@ -47,7 +47,7 @@ suspend fun <O> Stream<O>.drain(): Unit =
  * This a terminal operator, meaning this functions `suspend`s until the [Stream] finishes.
  * If any errors are raised while streaming, it's thrown from this `suspend` scope.
  */
-suspend fun <O> Stream<O>.firstOrNull(): O? =
+suspend fun <O> Stream<O>.first(): O? =
```

Comment:
This is inconsistent with Kotlin's standard library naming.

Reply:
You are right :) I was thinking if we are to add an extension to nullables like this:
```kotlin
fun <A> A?.orElse(f: () -> A): A = this ?: f()
```
Then, we wouldn't need to have `firstOrElse { }` and we could do: `first().orElse {  }`

I think Kotlin added `firstOrNull()` I think is because they already had `first()` that throws an error if there is no first element of the relevant collection... I think, from the get-go they should have returned a nullable value (and I think they probably would too) as it's safer.


All that said, I've made it consistent with stdlib :)

Reply:
> I think, from the get-go they should have returned a nullable value (and I think they probably would too) as it's safer.

I agree that there should be no `first` that throws an exception! but I think Kotlin added both consciously. Not 100% sure, but `require` and `check` is also often promoted by Kotlin.

> All that said, I've made it consistent with stdlib :)
👍 🎉